app/Policies/EngagementPolicy.php

```diff
@@ -6,18 +6,34 @@
 use App\Models\User;
 use Illuminate\Auth\Access\HandlesAuthorization;
 use Illuminate\Auth\Access\Response;
+use Illuminate\Support\Str;
 
 class EngagementPolicy
 {
     use HandlesAuthorization;
 
-    public function before(User $user): null|bool
+    public function before(User $user, string $ability): null|Response
     {
-        return $user->isAdministrator() ? true : null;
+        if ($user->isSuspended() && $ability !== 'view') {
+            return Response::deny(Str::markdown(
+                __('Your account has been suspended. Because of that, you do not have access to this page. Please contact us if you need further assistance.')
+                .contact_information()
+            ));
+        }
+
+        return null;
     }
 
     public function view(User $user, Engagement $engagement): Response
     {
+        if ($user->isSuspended() && $user->isAdministratorOf($engagement->project->projectable) && $engagement->isPublishable()) {
+            return Response::allow();
+        }
+
+        if ($user->isAdministrator() && $engagement->isPublishable()) {
+            return Response::allow();
+        }
+
         return
             ($user->individual || $user->organization || $user->regulated_organization)
```

Comment:
probably needs to allow admin users here. Can this pass for all authorized users, or is there a need to check all these different types?